updating getting started page and minor reorg of docs

choldgraf opened the pull request:
This PR is a general docs update with the following main points:

* Adding a CONTRIBUTING document similar to TLJH
* Updating the sample_repos script to organize them in topics since the list was getting super long
* Updates the getting_started section to have a more step-by-step intro
* Adds a "usecases" section for some sample usecases in the getting_started section
* Does some minor reorganization of the index page to try and make it easier to find the content that's relevant for people.

Here's a preview for the docs:

https://59-100564515-gh.circle-artifacts.com/0/html/index.html

lheagy commented on doc/generate_sample_repos.py:
```diff
@@ -0,0 +1,92 @@
+documentation page.
+"""
+
+from github import Github
```
How is this installed? I tried pypi and conda, and `github` was not found in either. Should we add it to the doc-requirements?

lheagy commented on doc/index.rst:
```diff
@@ -40,6 +38,20 @@ The following sections cover the basics of how to use a Binder service.
+How-To guides and Tutorials
+===========================
+
+The following sections some more in-depth topics in preparing and sharing
```
```suggestion
The following sections discuss some more in-depth topics on preparing and sharing
```

lheagy commented on doc/introduction.rst:
```diff
@@ -1,59 +1,135 @@
+your repository or in a ``binder/`` folder in the repository's root
+(i.e. ``myproject/binder/``).
+
+A Binder repository can be built by a BinderHub, which will generate a
```
Personally, I find this language a bit confusing. Even though you defined "Binder" as a repository that can be hosted on a BinderHub, my intuition uses "Binder" and "Binder Project" interchangeably here, so my first read of this sentence is that a "Binder repository" is a repository that is a part of the binder project, which is not the intention. 

Elsewhere, the language "your Binder" is used, which I would define as "my repo" + "a BinderHub". Would it work to define "a Binder" as "a Binder-ready repo" + "a BinderHub"?

For this sentence, would it be too strange to introduce the use of the word "Binderizable". So a "Binderizable repository" is a repo that can be built on BinderHub? Or alternatively using "Binder-ready" would clear up the language for me.  

Just my 2 cents, if you feel strongly about the language, I can work on adjusting my intuition :)

lheagy commented on doc/using.rst:
```diff
@@ -1,128 +1,111 @@
-  the repository's root.
-
-.. tip::
+For a more complete list of sample repositories for use with Binder, see the
```
Here, the word "Binder" is used as a service that connects to a repository (which is clear to me - it is just the use of Binder as it refers to a repo that messes with my head)

choldgraf replied:
Good point - we have always had a hard time disentangling these terms. Maybe something to put to rest at the dev meeting this year. In this PR I'll start using "Binder" to mean either "a service" or "the project", and I'll use binder-ready repository or binder repository for people's own work. That work?

lheagy replied:
"binder-ready repository" works for me. I find "Binder repository" potentially confusing, as it could easily be read as "Binder project repository" (e.g. it implies ownership by the Binder project, whereas "binder-ready" means it is ready to be connected to the Binder service)

lheagy commented on doc/using.rst:
```diff
@@ -1,128 +1,111 @@
-For a list of sample repositories for use with Binder, see the
-`Sample Binder Repositories <sample_repos.html>`_ page.
+Binder is also agnostic to the specific user interface that you'd like
+to share with your Binder. BinderHub will make some assumptions about the
```
This is another challenging one - "Binder" is being used as two different things in this sentence, first as a service, then as a "binderhub + repo". Can we exchange the first "Binder" with "BinderHub" ?

lheagy commented on doc/using.rst:
```diff
@@ -1,128 +1,111 @@
-``requirements.txt`` directive earlier.
+It is beyond the scope of the Binder documentation to cover the many
+ways that Binder can be used in an educational context. We recommend
+checking out the `Jupyter for Education guide <https://jupyter4edu.github.io/jupyter-edu-book/>`_
```
🎉

choldgraf commented on CONTRIBUTING.rst:
```diff
@@ -0,0 +1,216 @@
+.. note::
+
+   Heavily inspired by the
+   `django project's guidelines <https://docs.djangoproject.com/en/dev/internals/contributing/writing-documentation/>`_
```
This was originally modified for JupyterHub docs from @yuvipanda <3

choldgraf commented on CONTRIBUTING.rst:
```diff
@@ -0,0 +1,216 @@
+maximum flexibility. These files live in the top-level ``doc/`` directory of
+this repository.
+
+If you'd like to start contributing to our docs, get the development version of
```
as someone who was an editor for 5 years I really appreciate this suggestion

lheagy replied:
I have started an issue in #146

choldgraf commented:
OK I think I addressed your comments above - made some tweaks to the use of "binder" "binder-ready" etc, also added a point to discuss this at the team meeting because I think we still need to brainstorm the "right" way to go about this: https://github.com/jupyterhub/team-compass/issues/110#issuecomment-459411294

Ready for round 2 of comments! (or also this is good to go on my end)

choldgraf commented:
dependency added! I also modified environment.yml to use the `doc-requirements.txt` file in its install so that we're not repeating stuff.

re: the "image minimizer" stuff - how do you all feel about modifying the text to be a little less-strict? Instead we can say something like:

```
Choosing image size
===================

When adding images to the documentation, try to keep them as small as possible.
Larger images make the site load more slowly on browsers, and may make the site
inaccessible for people with a slow internet connection.

If you're adding screenshots, make the size of your shot as small as possible.
If you're uploading large images, consider using an image optimizer in order
to reduce its size.

For example, for PNG files, use OptiPNG and
AdvanceCOMP's ``advpng``:

.. code-block:: console

   $ cd docs
   $ optipng -o7 -zm1-9 -i0 -strip all `find . -type f -not -path "./_build/*" -name "*.png"`
   $ advpng -z4 `find . -type f -not -path "./_build/*" -name "*.png"`

This is based on OptiPNG version 0.7.5. Older versions may complain about the
``--strip all`` option being lossy.


```

I really doubt people will follow this by default and I also doubt that team-members want to block a PR because it hasn't had `optipng` run on it. It adds an extra barrier to contribution for the sake of (probably minimal) speed optimization. Maybe this language will still get the point across ("use small images please!") without making it a "rule"

lheagy commented:
Looks good @choldgraf! and 👍on the text for the image size

yuvipanda commented:
@choldgraf when you have time, would love a backport of the text around image optimization back to TLJH :)

choldgraf commented:
OK, updated images text (and check the TLJH repo for a PR @yuvipanda !) and requirements.

lheagy commented:
Merged :)
On Fri, Feb 1, 2019 at 2:23 PM Chris Holdgraf <notifications@github.com>
wrote:

> (just a note that this is GTG from my perspective)
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/jupyterhub/binder/pull/144#issuecomment-459888752>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AGES1HwJ7G_VpBgjZR6bPrO-edNIBzaCks5vJL58gaJpZM4aa0zH>
> .
>
